chapters/ch55.md

```diff
@@ -40,36 +40,37 @@ Xe tự lái của bạn sử dụng pipeline như trên. Bạn sẽ sử dụng
 
 1. Thử điều chỉnh thủ công kết quả đầu ra của thành phần A (phát hiện xe hơi) sao cho "hoàn hảo" (ví dụ, cho xe biết vị trí của những chiếc xe khác). Sau đó tiếp tục chạy phần còn lại của pipeline gồm có B, C, nhưng cho phép C (hoạch định đường đi) sử dụng đầu ra đã hoàn hảo của A. Nếu thuật toán hoạch định đường đi cho xe tốt hơn, điều đó cho thấy rằng, kết quả cuối cùng của thuật toán tự lái sẽ được cải thiện nếu mà A trả về kết quả tốt hơn.  Như vậy, bạn có thể quy lỗi cho A. Nếu không, ta tiếp tục bước 2. 
 
 > 2. Try manually modifying B (detect pedestrian)’s output to be the "perfect" output for B. If the algorithm now gives a correct output, then attribute the error to component B. Otherwise, go on to Step 3.
 
-2. Thử điều chỉnh thủ công kết quả đầu ra ở công đoạn B (phát hiện người đi bộ) cho "hoàn hảo". Nếu thuật toán cho ra kết quả đầu ra cuối cùng chính xác, ta có thể quy lỗi cho B. Ngược lại, ta tiến hành bước 3.
+2. Thử điều chỉnh thủ công kết quả đầu ra ở công đoạn B (phát hiện người đi bộ) cho "hoàn hảo". Nếu thuật toán cho kết quả đầu ra cuối cùng chính xác, ta có thể quy lỗi cho B. Ngược lại, ta tiến hành bước 3.
+
 > 3. Attribute the error to component C.
 
 3. Quy lỗi cho thành phần C. 
 
 > The components of an ML pipeline should be ordered according to a Directed Acyclic Graph (DAG), meaning that you should be able to compute them in some fixed left-to-right order, and later components should depend only on earlier components’ outputs. So long as the mapping of the components to the A->B->C order follows the DAG ordering, then the error analysis will be fine. You might get slightly different results if you swap A and B:
 
-Các thành phần của một mô hình học máy dạng pipeline nên được sắp xếp theo đồ thị có hướng không chu trình (DAG), nghĩa là bạn có thể tính toán chúng theo thứ tự cố định từ trái sang phải nào đó, và các thành phần sau chỉ nên phụ thuộc vào đầu ra của các thành phần trước đó. Miễn là việc xâu chuỗi các thành phần theo thứ tự A->B->C tuân thủ theo quy tắc DAG, việc phân tích lỗi sẽ tốt. Bạn có thể nhận được các kết quả hơi khác nhau nếu hoán chuyển vị trí của A và B cho nhau như sau:
+Các thành phần của một mô hình học máy dạng pipeline nên được sắp xếp theo Đồ thị có hướng không chu trình (Directed Acyclic Graph - DAG), nghĩa là bạn có thể tính toán chúng theo thứ tự cố định từ trái sang phải nào đó, và các thành phần sau chỉ nên phụ thuộc vào đầu ra của các thành phần trước đó. Miễn là việc xâu chuỗi các thành phần theo thứ tự A->B->C tuân thủ theo quy tắc DAG, thì việc phân tích lỗi sẽ diễn ra tốt đẹp. Bạn có thể nhận được các kết quả hơi khác nhau nếu hoán chuyển vị trí của A và B cho nhau như sau:
```

Comment:
```suggestion
Các thành phần của một mô hình học máy dạng pipeline nên được sắp xếp theo Đồ thị có hướng không chu trình (Directed Acyclic Graph - DAG), nghĩa là bạn có thể tính toán chúng theo thứ tự cố định từ trái sang phải nào đó, và các thành phần sau chỉ nên phụ thuộc vào đầu ra của các thành phần trước đó. Miễn là việc xâu chuỗi các thành phần theo thứ tự A->B->C tuân thủ theo quy tắc DAG, thì việc phân tích lỗi sẽ diễn ra tốt đẹp. Bạn có thể nhận được các kết quả hơi khác nhau nếu hoán chuyển vị trí của A và B cho nhau như sau:
```

Từ Directed Acyclic Graph được thảo luận ở đâu nhỉ? Mình thấy cách dịch hiện tại chưa chuẩn nghĩa lắm

Reply:
@ngcthuong 
Từ này được dịch theo các khái niệm về "độ thị có hướng" và "chu trình" trong đồ thị.

Phần "độ thị có hướng" có lẽ không cần bàn luận nhiều vì [đã được dùng phổ biến](https://www.google.com/search?q=%C4%91%E1%BB%93+th%E1%BB%8B+c%C3%B3+h%C6%B0%E1%BB%9Bng&oq=%C4%91%E1%BB%93+th%E1%BB%8B+c%C3%B3+h%C6%B0%E1%BB%9Bng+&aqs=chrome..69i57j0l7.3637j0j9&sourceid=chrome&ie=UTF-8).

Khái niệm "chu trình" trong đồ thị cũng [phổ biến](https://www.google.com/search?ei=ICv3Xfi7FfmS0PEPx8aD6A8&q=chu+tr%C3%ACnh+trong+%C4%91%E1%BB%93+th%E1%BB%8B+&oq=chu+tr%C3%ACnh+trong+%C4%91%E1%BB%93+th%E1%BB%8B+&gs_l=psy-ab.3..0i19j0i22i30i19l6.2146.3614..3768...0.0..0.113.1168.10j3......0....1..gws-wiz.......0i22i30j0.3kgwlhy86W8&ved=0ahUKEwj4y7nuy7nmAhV5CTQIHUfjAP0Q4dUDCAs&uact=5), nó là một đường khép kín đi qua một tập con các đỉnh của đồ thị.

Việc dịch "đồ thị có hướng không chu trình" theo mình là hoàn toàn tự nhiên.